include/revng/Pipes/Isolate.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/EarlyFunctionAnalysis/ABIDetectionPass.h"
+#include "revng/FunctionIsolation/InvokeIsolatedFunctions.h"
+#include "revng/FunctionIsolation/IsolateFunctions.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+
+namespace pipeline {
+
+struct ABIDetectPipe {
+  static constexpr auto Name = "ABIDetect";
+
+  std::vector<ContractGroup> getContract() const { return {}; }
```

Comment:
Is there a reason to define this? Can't the user use `PureLLVMPass`?
We need to decide what to do in these cases. I'm not sure we want to put the burden on the user to define a class for each pass.

Reply:
the problem would be that at the moment the parser rules to load purellvmpasses or non pure ones are unrelated, that is the names do not pollute each other, so you could not load a pure llvm pass inside a non pure llvm pipe. it would require some working to make sure that they work together.

Reply:
Mmmmh, this is rather unexpected. I'm afraid we need to fix this. In particular approaching the back of the decompilation pipeline we mix heavy LLVM analysis and our passes, it's not very nice to discard all the analysis results for infra limitations.

Can you see how much effort it'd be and decide whether it fits within the scope of this PR=